### PR TITLE
Provide a handle for the ServiceProvider stub used by the application…

### PR DESCRIPTION
**Pull request by jason-simmons**
… to expose services to the embedder

### CODE REVIEW

File: sky/shell/ui/internals.cc [hunk 1]
```diff
@@ -42,6 +42,8 @@ void TakeServicesProvidedByEmbedder(Dart_NativeArguments args) {
 
 void TakeServicesProvidedToEmbedder(Dart_NativeArguments args) {
   Dart_SetIntegerReturnValue(args, 0);
```
Comment (abarth) [inline sky/shell/ui/internals.cc]: We can remove this line.

File: sky/shell/ui/internals.h [hunk 2]
```diff
@@ -46,6 +47,12 @@ class Internals
   mojo::ServiceProviderImpl service_provider_impl_;
   mojo::ServiceProviderPtr platform_service_provider_;
 
+  // Proxy for a ServiceProvider that exposes services from the application to
+  // the embedder.
+  mojo::ServiceProviderPtr application_service_provider_;
+  mojo::InterfaceRequest<mojo::ServiceProvider>
+      application_service_provider_request_;
```
Comment (abarth) [inline sky/shell/ui/internals.h]: The names for these things get kind of crazy.  Maybe call this services_provided_by_embedder_ ?

File: sky/shell/ui/internals.h [hunk 3]
```diff
@@ -46,6 +47,12 @@ class Internals
   mojo::ServiceProviderImpl service_provider_impl_;
   mojo::ServiceProviderPtr platform_service_provider_;
 
+  // Proxy for a ServiceProvider that exposes services from the application to
+  // the embedder.
+  mojo::ServiceProviderPtr application_service_provider_;
```
Comment (abarth) [inline sky/shell/ui/internals.h]: There's no reason to hold on to this object.  We can just make it a local variable in the constructor that goes out of scope immediately.

**Comment by abarth** [comment]
LGTM

File: sky/shell/ui/internals.cc (hunk 1 above)
Comment (jason-simmons) [inline sky/shell/ui/internals.cc]: done

File: sky/shell/ui/internals.h (hunk 2 above)
Comment (jason-simmons) [inline sky/shell/ui/internals.h]: done

File: sky/shell/ui/internals.h (hunk 3 above)
Comment (jason-simmons) [inline sky/shell/ui/internals.h]: done

**Comment by jason-simmons** [comment]
PTAL

**Comment by abarth** [comment]
LGTM

**Pull request merged**